invenio_records_resources/services/records/service.py

```diff
@@ -288,6 +291,71 @@ def read(self, id_, identity):
             links_tpl=self.links_item_tpl,
         )
 
+    def _read_many(self, identity, es_query, fields=None, record_cls=None,
+                   search_opts=None, **kwargs):
+        """Search for records matching the ids."""
+        # We use create_search() to avoid the overhead of aggregations etc
+        # being added to the query with using search_request().
+        search = self.create_search(
+            identity=identity,
+            record_cls=record_cls or self.record_cls,
+            search_opts=search_opts or self.config.search,
+            permission_action='search',
+        )
+
+        # Fetch only certain fields - explicitly add internal system fields
+        # required to use the result list to dump the output.
+        if fields:
+            dumper_fields = [
+                "uuid", "version_id", "created", "updated", "expires_at"]
+            fields = fields + dumper_fields
+            # ES 7.11+ supports a more efficient way of fetching only certain
+            # fields using the "fields"-option to a query. However, ES 6 and
+            # ES 7 versions does not support it, so we use the source filtering
+            # method instead for now.
+            search = search.source(fields)
+
+        search_result = search.query(es_query).execute()
+
+        return search_result
+
+    def read_many(self, identity, ids, fields=None, **kwargs):
+        """Search for records matching the ids."""
+        clauses = []
+        for id_ in ids:
+            clauses.append(Q('term', **{"id": id_}))
+        query = Q('bool', minimum_should_match=1, should=clauses)
+
+        results = self._read_many(identity, query, fields, **kwargs)
+
+        return self.result_list(self, identity, results)
+
+    def read_all(self, identity, fields, cache=True, **kwargs):
```

Comment:
Just another wrinkle I ran into while producing the resource type choices on the upload page:

There we need, for each resource type:

- id
- type name
- subtype name

type name and subtype name are found in the "props" . I am not sure how we want to deal with those. For now, in invenio-app-rdm I generate the vocabulary list from the database. :shrug:

Reply:
would passing `props.type` and `props.subtype` in the `fields` parameter work? sort of like [here](https://github.com/inveniosoftware/invenio-records-resources/pull/191/files#diff-0be63bb442d15c429661d7b94b34276b9bae5cf6c81562f2699b78e3bdfafe55R77)